[CI] JdbcSecurityIT testQueryDocumentExcluded failing

I think there was a similar occurrence on June 1st: https://gradle-enterprise.elastic.co/s/kmwjw4dcwdppk

**Build scan:**
https://gradle-enterprise.elastic.co/s/3gt7diubtz7mu/tests/:x-pack:plugin:sql:qa:server:security:with-ssl:integTest/org.elasticsearch.xpack.sql.qa.security.JdbcSecurityIT/testQueryDocumentExcluded

**Reproduction line:**
`./gradlew ':x-pack:plugin:sql:qa:server:security:with-ssl:integTest' --tests "org.elasticsearch.xpack.sql.qa.security.JdbcSecurityIT.testQueryDocumentExcluded" -Dtests.seed=68E003C7EDCA036 -Dtests.locale=zh -Dtests.timezone=Australia/Yancowinna -Druntime.java=11`

**Applicable branches:**
master

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.sql.qa.security.JdbcSecurityIT&tests.test=testQueryDocumentExcluded

**Failure excerpt:**
```
java.lang.AssertionError: Some checkers [0, 1, 2, 3] didn't match any logs. All logs:  none!
Remaining logs:  none!

  at __randomizedtesting.SeedInfo.seed([68E003C7EDCA036:B17D9F735B81E0BA]:0)
  at org.junit.Assert.fail(Assert.java:88)
  at org.elasticsearch.xpack.sql.qa.security.SqlSecurityTestCase$AuditLogAsserter.lambda$assertLogs$3(SqlSecurityTestCase.java:677)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:965)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:938)
  at org.elasticsearch.xpack.sql.qa.security.SqlSecurityTestCase$AuditLogAsserter.assertLogs(SqlSecurityTestCase.java:575)
  at org.elasticsearch.xpack.sql.qa.security.SqlSecurityTestCase.testQueryDocumentExcluded(SqlSecurityTestCase.java:362)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:566)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:829)

```

Another one: https://gradle-enterprise.elastic.co/s/6kgjg3tj2tl5u/tests/failed

The above build scan has a few other test failures. The error message is:
> java.lang.AssertionError: Previous test had an audit-related failure. All subsequent audit related assertions are bogus because we can't guarantee that we fully cleaned up after the last test.	

Seems like a test setup issue. So it might be cascading failures due to failure of `testQueryDocumentExcluded`.

Leaving here some snippets from various failures:

```
  1> [2021-07-19T17:59:59,017][INFO ][o.e.x.s.q.s.JdbcSecurityIT] [testShowTablesWithLimitedAccessUnaccessableIndex] after test
  1> [2021-07-19T17:59:59,036][INFO ][o.e.x.s.q.s.JdbcSecurityIT] [testShowTablesWorksAsFullAccess] before test
  1> [2021-07-19T17:59:59,959][INFO ][o.e.x.s.q.s.JdbcSecurityIT] [testShowTablesWorksAsFullAccess] after test
  1> [2021-07-19T17:59:59,977][INFO ][o.e.x.s.q.s.JdbcSecurityIT] [testQueryDocumentExcluded] before test
  1> [2021-07-19T18:00:10,708][WARN ][o.e.x.s.q.s.JdbcSecurityIT] [testQueryDocumentExcluded] Failed to find an audit log. Skipping remaining tests in this class after this the missing auditlogs could turn up later.
  1> [2021-07-19T18:00:10,902][INFO ][o.e.x.s.q.s.JdbcSecurityIT] [testQueryDocumentExcluded] after test
  2> REPRODUCE WITH: ./gradlew ':x-pack:plugin:sql:qa:server:security:with-ssl:integTest' --tests "org.elasticsearch.xpack.sql.qa.security.JdbcSecurityIT.testQueryDocumentExcluded" -Dtests.seed=A7CC4DDCA50FB1EA -Dtests.locale=hu -Dtests.timezone=Pacific/Easter -Druntime.java=11
  2> java.lang.AssertionError: Some checkers [0, 1, 2, 3] didn't match any logs. All logs:  none!
    Remaining logs:  none!
        at __randomizedtesting.SeedInfo.seed([A7CC4DDCA50FB1EA:103FD2938052F166]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.elasticsearch.xpack.sql.qa.security.SqlSecurityTestCase$AuditLogAsserter.lambda$assertLogs$3(SqlSecurityTestCase.java:677)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:973)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:946)
        at org.elasticsearch.xpack.sql.qa.security.SqlSecurityTestCase$AuditLogAsserter.assertLogs(SqlSecurityTestCase.java:575)
        at org.elasticsearch.xpack.sql.qa.security.SqlSecurityTestCase.testQueryDocumentExcluded(SqlSecurityTestCase.java:362)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
```

```
  1> [2021-06-07T01:59:59,200][INFO ][o.e.x.s.q.s.JdbcSecurityIT] [testShowTablesWithNoAccess] before test
  1> [2021-06-07T01:59:59,957][INFO ][o.e.x.s.q.s.JdbcSecurityIT] [testShowTablesWithNoAccess] after test
  1> [2021-06-07T01:59:59,984][INFO ][o.e.x.s.q.s.JdbcSecurityIT] [testScrollWithSingleFieldExcepted] before test
  1> [2021-06-07T02:00:10,368][WARN ][o.e.x.s.q.s.JdbcSecurityIT] [testScrollWithSingleFieldExcepted] Failed to find an audit log. Skipping remaining tests in this class after this the missing auditlogs could turn up later.
  1> [2021-06-07T02:00:10,788][INFO ][o.e.x.s.q.s.JdbcSecurityIT] [testScrollWithSingleFieldExcepted] after test
  2> REPRODUCE WITH: gradlew ':x-pack:plugin:sql:qa:server:security:without-ssl:integTest' --tests "org.elasticsearch.xpack.sql.qa.security.JdbcSecurityIT.testScrollWithSingleFieldExcepted" -Dtests.seed=B47328DD358C3E5D -Dtests.locale=en-AU -Dtests.timezone=Etc/GMT-2 -Druntime.java=11  2> 
  2> java.lang.AssertionError: Some checkers [0, 1, 2, 3, 4, 5, 6, 7] didn't match any logs. All logs:  none!
    Remaining logs:  none!
        at __randomizedtesting.SeedInfo.seed([B47328DD358C3E5D:619AC14499317F61]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.elasticsearch.xpack.sql.qa.security.SqlSecurityTestCase$AuditLogAsserter.lambda$assertLogs$3(SqlSecurityTestCase.java:677)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:973)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:946)
        at org.elasticsearch.xpack.sql.qa.security.SqlSecurityTestCase$AuditLogAsserter.assertLogs(SqlSecurityTestCase.java:575)
        at org.elasticsearch.xpack.sql.qa.security.SqlSecurityTestCase.testScrollWithSingleFieldExcepted(SqlSecurityTestCase.java:340)
```

```
  1> [2021-06-08T01:59:59,129][INFO ][o.e.x.s.q.s.JdbcSecurityIT] [testDescribeWithWrongAccess] before test
  1> [2021-06-08T01:59:59,624][INFO ][o.e.x.s.q.s.JdbcSecurityIT] [testDescribeWithWrongAccess] after test
  1> [2021-06-08T01:59:59,639][INFO ][o.e.x.s.q.s.JdbcSecurityIT] [testScrollWithSingleFieldExcepted] before test
  1> [2021-06-08T02:00:10,337][WARN ][o.e.x.s.q.s.JdbcSecurityIT] [testScrollWithSingleFieldExcepted] Failed to find an audit log. Skipping remaining tests in this class after this the missing auditlogs could turn up later.
  1> [2021-06-08T02:00:10,429][INFO ][o.e.x.s.q.s.JdbcSecurityIT] [testScrollWithSingleFieldExcepted] after test
  2> REPRODUCE WITH: ./gradlew ':x-pack:plugin:sql:qa:server:security:with-ssl:integTest' --tests "org.elasticsearch.xpack.sql.qa.security.JdbcSecurityIT.testScrollWithSingleFieldExcepted" -Dtests.seed=E18AAF3CA97F64D7 -Dtests.locale=es-CR -Dtests.timezone=Arctic/Longyearbyen -Druntime.java=17
  2> java.lang.AssertionError: Some checkers [0, 1, 2, 3, 4, 5, 6, 7] didn't match any logs. All logs:  none!
    Remaining logs:  none!
        at __randomizedtesting.SeedInfo.seed([E18AAF3CA97F64D7:346346A505C225EB]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.elasticsearch.xpack.sql.qa.security.SqlSecurityTestCase$AuditLogAsserter.lambda$assertLogs$3(SqlSecurityTestCase.java:677)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:973)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:946)
        at org.elasticsearch.xpack.sql.qa.security.SqlSecurityTestCase$AuditLogAsserter.assertLogs(SqlSecurityTestCase.java:575)
        at org.elasticsearch.xpack.sql.qa.security.SqlSecurityTestCase.testScrollWithSingleFieldExcepted(SqlSecurityTestCase.java:340)
```

```
  1> [2021-07-19T17:59:58,091][INFO ][o.e.x.s.q.s.JdbcSecurityIT] [testShowTablesWithLimitedAccessUnaccessableIndex] before test
  1> [2021-07-19T17:59:59,017][INFO ][o.e.x.s.q.s.JdbcSecurityIT] [testShowTablesWithLimitedAccessUnaccessableIndex] after test
  1> [2021-07-19T17:59:59,036][INFO ][o.e.x.s.q.s.JdbcSecurityIT] [testShowTablesWorksAsFullAccess] before test
  1> [2021-07-19T17:59:59,959][INFO ][o.e.x.s.q.s.JdbcSecurityIT] [testShowTablesWorksAsFullAccess] after test
  1> [2021-07-19T17:59:59,977][INFO ][o.e.x.s.q.s.JdbcSecurityIT] [testQueryDocumentExcluded] before test
  1> [2021-07-19T18:00:10,708][WARN ][o.e.x.s.q.s.JdbcSecurityIT] [testQueryDocumentExcluded] Failed to find an audit log. Skipping remaining tests in this class after this the missing auditlogs could turn up later.
  1> [2021-07-19T18:00:10,902][INFO ][o.e.x.s.q.s.JdbcSecurityIT] [testQueryDocumentExcluded] after test
  2> REPRODUCE WITH: ./gradlew ':x-pack:plugin:sql:qa:server:security:with-ssl:integTest' --tests "org.elasticsearch.xpack.sql.qa.security.JdbcSecurityIT.testQueryDocumentExcluded" -Dtests.seed=A7CC4DDCA50FB1EA -Dtests.locale=hu -Dtests.timezone=Pacific/Easter -Druntime.java=11
  2> java.lang.AssertionError: Some checkers [0, 1, 2, 3] didn't match any logs. All logs:  none!
    Remaining logs:  none!
        at __randomizedtesting.SeedInfo.seed([A7CC4DDCA50FB1EA:103FD2938052F166]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.elasticsearch.xpack.sql.qa.security.SqlSecurityTestCase$AuditLogAsserter.lambda$assertLogs$3(SqlSecurityTestCase.java:677)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:973)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:946)
        at org.elasticsearch.xpack.sql.qa.security.SqlSecurityTestCase$AuditLogAsserter.assertLogs(SqlSecurityTestCase.java:575)
        at org.elasticsearch.xpack.sql.qa.security.SqlSecurityTestCase.testQueryDocumentExcluded(SqlSecurityTestCase.java:362)
```

--- a/x-pack/plugin/sql/qa/server/security/build.gradle
+++ b/x-pack/plugin/sql/qa/server/security/build.gradle
@@ -50,7 +50,7 @@ subprojects {
     nonInputProperties.systemProperty 'tests.audit.logfile',
       "${-> testClusters.integTest.singleNode().getAuditLog()}"
     nonInputProperties.systemProperty 'tests.audit.yesterday.logfile',
-      "${-> testClusters.integTest.singleNode().getAuditLog().getParentFile()}/integTest_audit-${new Date().format('yyyy-MM-dd')}.json"
+      "${-> testClusters.integTest.singleNode().getAuditLog().getParentFile()}/integTest_audit-${new Date().format('yyyy-MM-dd')}-1.json.gz"
   }
 
   tasks.named("testingConventions").configure { enabled = false }

--- a/x-pack/plugin/sql/qa/server/security/src/test/java/org/elasticsearch/xpack/sql/qa/security/CliSecurityIT.java
+++ b/x-pack/plugin/sql/qa/server/security/src/test/java/org/elasticsearch/xpack/sql/qa/security/CliSecurityIT.java
@@ -28,19 +28,15 @@ import static org.hamcrest.Matchers.startsWith;
 public class CliSecurityIT extends SqlSecurityTestCase {
 
     @Override
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/76806")
     public void testDescribeWorksAsFullAccess() {}
 
     @Override
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/76806")
     public void testQuerySingleFieldGranted() {}
 
     @Override
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/76806")
     public void testScrollWithSingleFieldExcepted() {}
 
     @Override
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/76806")
     public void testQueryWorksAsAdmin() {}
 
     static SecurityConfig adminSecurityConfig() {

--- a/x-pack/plugin/sql/qa/server/security/src/test/java/org/elasticsearch/xpack/sql/qa/security/SqlSecurityTestCase.java
+++ b/x-pack/plugin/sql/qa/server/security/src/test/java/org/elasticsearch/xpack/sql/qa/security/SqlSecurityTestCase.java
@@ -28,7 +28,11 @@ import org.junit.AfterClass;
 import org.junit.Before;
 
 import java.io.BufferedReader;
+import java.io.FileInputStream;
 import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
@@ -44,6 +48,7 @@ import java.util.Map;
 import java.util.TreeMap;
 import java.util.function.Function;
 import java.util.stream.Collectors;
+import java.util.zip.GZIPInputStream;
 
 import static java.util.Arrays.asList;
 import static java.util.Collections.singletonMap;
@@ -587,8 +592,12 @@ public abstract class SqlSecurityTestCase extends ESRestTestCase {
                             if (localAuditFileRolledOver == false && Files.exists(ROLLED_OVER_AUDIT_LOG_FILE)) {
                                 // once the audit file rolled over, it will stay like this
                                 auditFileRolledOver = true;
+                                // unzip the file
+                                InputStream fileStream = new FileInputStream(ROLLED_OVER_AUDIT_LOG_FILE.toFile());
+                                InputStream gzipStream = new GZIPInputStream(fileStream);
+                                Reader decoder = new InputStreamReader(gzipStream, StandardCharsets.UTF_8);
                                 // the order in the array matters, as the readers will be used in that order
-                                logReaders[0] = Files.newBufferedReader(ROLLED_OVER_AUDIT_LOG_FILE, StandardCharsets.UTF_8);
+                                logReaders[0] = new BufferedReader(decoder);
                             }
                             logReaders[1] = Files.newBufferedReader(AUDIT_LOG_FILE, StandardCharsets.UTF_8);
                             return null;

--- a/x-pack/plugin/sql/qa/server/security/src/test/java/org/elasticsearch/xpack/sql/qa/security/SqlSecurityTestCase.java
+++ b/x-pack/plugin/sql/qa/server/security/src/test/java/org/elasticsearch/xpack/sql/qa/security/SqlSecurityTestCase.java
@@ -28,7 +28,6 @@ import org.junit.AfterClass;
 import org.junit.Before;
 
 import java.io.BufferedReader;
-import java.io.FileInputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.InputStreamReader;
@@ -593,8 +592,7 @@ public abstract class SqlSecurityTestCase extends ESRestTestCase {
                                 // once the audit file rolled over, it will stay like this
                                 auditFileRolledOver = true;
                                 // unzip the file
-                                InputStream fileStream = new FileInputStream(ROLLED_OVER_AUDIT_LOG_FILE.toFile());
-                                InputStream gzipStream = new GZIPInputStream(fileStream);
+                                InputStream gzipStream = new GZIPInputStream(Files.newInputStream(ROLLED_OVER_AUDIT_LOG_FILE));
                                 Reader decoder = new InputStreamReader(gzipStream, StandardCharsets.UTF_8);
                                 // the order in the array matters, as the readers will be used in that order
                                 logReaders[0] = new BufferedReader(decoder);

